From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sat, 22 Mar 2026 00:00:00 +0000
Subject: [PATCH] Add absl/base:no_destructor for system com_google_absl

Abseil LTS 20250127.1 exposes no_destructor; the slim system base BUILD
did not, breaking XLA shape_util and other deps.
---
 third_party/xla/third_party/absl/system.absl.base.BUILD | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/third_party/xla/third_party/absl/system.absl.base.BUILD b/third_party/xla/third_party/absl/system.absl.base.BUILD
index 6534793637c..a1b2c3d4e5f 100644
--- a/third_party/xla/third_party/absl/system.absl.base.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.base.BUILD
@@ -41,6 +41,14 @@ cc_library(
 )
 
 cc_library(
+    name = "no_destructor",
+    deps = [
+        ":config",
+        ":nullability",
+    ],
+)
+
+cc_library(
     name = "raw_logging_internal",
     linkopts = [
         "-labsl_raw_logging_internal",
